ext/session/session.c

```diff
@@ -2711,6 +2723,22 @@ static const zend_function_entry session_functions[] = {
 
 /* {{{ SessionHandlerInterface functions[]
 */
+static const zend_function_entry php_session_save_handler_iface_functions[] = {
+	PHP_ABSTRACT_ME(SessionSaveHandlerInterface, open, arginfo_session_class_open)
```

Comment:
Since session is closely related to authentication and privacy, it must be as secure as possible by default.  IMO.

For example,  shopping sites that allows to add product without authentication can leak cart contents to attacker without session ID validation. When attackers has physical access device, (e.g. colleague, shared pc, etc), it's easy to set session ID for attack. Attackers do not even have to keep session ID alive. There are other attack scenarios.

One may argue "these risk are negligible", but this isn't how we secure softwares. We should remove risks that can be removed easily. 

Currently, validating session ID has extra overhead, but I'm going to propose remove this overhead. So there will be no overhead at all soon.